ios/CodePush/CodePush.m

```diff
@@ -455,9 +455,12 @@ - (void)rollbackPackage
 {
     NSError *error;
     NSDictionary *failedPackage = [CodePushPackage getCurrentPackage:&error];
-
-    // Write the current package's metadata to the "failed list"
-    [self saveFailedUpdate:failedPackage];
+    if (error) {
```

Comment:
It's already checked below in the `else if`. The current implementation is much better because it does both: prints the error if one exists, and if it doesnt (which signifies a problem with the method's contract) that gets handled too.